addons/im_livechat/static/src/new/core/message_service_patch.js

```diff
@@ -14,4 +14,10 @@ patch(MessageService.prototype, "im_livechat", {
         }
         return message;
     },
+    update(message, data) {
+        this._super(message, data);
+        if (!message.author && this.store.self) {
+            message.author = this.store.self;
+        }
+    },
```

Comment:
This looks like a hack, it would be wise to put a comment so we know it's intentional (and it clarifies the assumptions that are true at the moment of writing, in case it becomes wrong or undesirable in the future)

Reply:
Hack removed :)